src/particleRepresentation/container/ParticleContainer.h

```diff
@@ -0,0 +1,60 @@
+//
+// Created by daniel on 22.05.24.
+//
+
+#pragma once
+#include "particleRepresentation/particle/Particle.h"
+
+/**
+ * Abstract base class for a particle container. Each type of a particle container should extend this class.
+ */
+class ParticleContainer {
+public:
+
+    /**
+     * @brief Add one particle to this container.
+     *
+     * @param p Particle to be added to this container.
+     *
+     * Add one particle to this container. It will be appended.
+     */
+
+    virtual void add(Particle& p) = 0;
+
+    /**
+     * @brief Get the number of particles stored in this container.
+     *
+     * @return Number of particles stored in this container.
+     */
+    virtual size_t size() = 0;
```

Comment:
```suggestion
    virtual size_t size() const = 0;
```
Just as an example here. You could start marking some class methods as const, meaning they guarantee to not change the object, when being called.

Reply:
I saw later, that you did this almost everywhere else, so this is just a small reminder then ;)